src/ripple/rpc/handlers/AccountObjects.cpp

```diff
@@ -0,0 +1,190 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012-2014 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/rpc/impl/Tuning.h>
+#include <ripple/protocol/Indexes.h>
+
+namespace ripple {
+
+/** General RPC command that can retrieve objects in the account root.    
+    {
+      account: <account>|<account_public_key>
+      account_index: <integer> // optional, defaults to 0
+      ledger_hash: <string> // optional
+      ledger_index: <string | unsigned integer> // optional
+      limit: <integer> // optional
+      marker: <opaque> // optional, resume previous query
+    }
+*/
+Json::Value doAccountObjects (RPC::Context& context)
+{
+    auto const& params (context.params);
+    if (! params.isMember (jss::account))
+        return RPC::missing_field_error (jss::account);
+
+    Ledger::pointer ledger;
+    Json::Value result (RPC::lookupLedger (params, ledger, context.netOps));
+    if (ledger == nullptr)
+        return result;
+
+    RippleAddress rippleAddress;    
+    {
+        bool bIndex;
+        std::string const strIdent (params[jss::account].asString ());
+        int const iIndex (context.params.isMember (jss::account_index)
+            ? context.params[jss::account_index].asUInt () : 0);
+        Json::Value const jv(RPC::accountFromString(ledger, rippleAddress, bIndex,
+            strIdent, iIndex, false, context.netOps));
+        if (! jv.empty ())
+        {
+            for (Json::Value::const_iterator it (jv.begin ()); it != jv.end (); ++it)
```

Comment:
Unfortunately auto will not resolve to a Json::Value::const_iterator, otherwise I would.